homeassistant/components/smhi/config_flow.py

```diff
@@ -48,6 +48,10 @@ async def async_step_user(
             if is_ok:
                 name = slugify(user_input[CONF_NAME])
                 if not self._name_in_configuration_exists(name):
```

Comment:
This check is weird, especially if we're setting a unique id for the config entry.

Reply:
Didn't necessarily want to touch the "legacy" stuff in this just yet. But otherwise I agree with you, could understand why testing for the name but not the location itself.
Anyway can we leave that for next step?